pkg/cgroups/cgroups.go

```diff
@@ -52,9 +53,12 @@ func CheckOrMountCgrpFS(mapRoot string) {
 		if mapRoot == "" {
 			mapRoot = cgroupRoot
 		}
-		err := cgrpCheckOrMountLocation(mapRoot)
-		// Failed cgroup2 mount is not a fatal error, sockmap will be disabled however
-		if err == nil {
+
+		if err := cgrpCheckOrMountLocation(mapRoot); err != nil {
+			if option.Config.EnableHostReachableServices {
+				log.WithError(err).Warnf("Failed to mount cgroupv2. Socket-based LB will not work.")
```

Comment:
nit: No need to `Warnf`, since `Warnf` runs the logic to parse the string to find arguments and then handles the arguments. `Warn()` should be sufficient.

```suggestion
				log.WithError(err).Warn("Failed to mount cgroupv2. Socket-based LB will not work.")
```